subprojects/platform-base/src/main/java/org/gradle/api/plugins/BasePlugin.java

```diff
@@ -80,24 +80,19 @@ private void configureBuildConfigurationRule(Project project) {
     }
 
     private void configureConfigurations(final Project project) {
-        ConfigurationContainer configurations = project.getConfigurations();
+        RoleBasedConfigurationContainerInternal configurations = (RoleBasedConfigurationContainerInternal) project.getConfigurations();
         ((ProjectInternal)project).getInternalStatus().convention("integration");
 
-        final Configuration archivesConfiguration = configurations.maybeCreate(Dependency.ARCHIVES_CONFIGURATION).
+        final Configuration archivesConfiguration = configurations.maybeCreateWithRole(Dependency.ARCHIVES_CONFIGURATION, ConfigurationRoles.INTENDED_CONSUMABLE, false, false).
             setDescription("Configuration for archive artifacts.");
 
-        final Configuration defaultConfiguration =  configurations.maybeCreate(Dependency.DEFAULT_CONFIGURATION).
+        final Configuration defaultConfiguration =  configurations.maybeCreateWithRole(Dependency.DEFAULT_CONFIGURATION, ConfigurationRoles.INTENDED_CONSUMABLE, false, false).
             setDescription("Configuration for default artifacts.");
 
         final DefaultArtifactPublicationSet defaultArtifacts = project.getExtensions().create(
             "defaultArtifacts", DefaultArtifactPublicationSet.class, archivesConfiguration.getArtifacts()
         );
 
-        archivesConfiguration.setCanBeResolved(false);
-        defaultConfiguration.setCanBeResolved(false);
-        ((ConfigurationInternal) archivesConfiguration).setCanBeDeclaredAgainst(false);
```

Comment:
We will be reverting this line due to the 8.0 regression which was just reported. We may need to avoid using the builders for this configuration or alternatively use a custom role.